tests/test_notes.py

```diff
@@ -206,21 +228,70 @@ def test_chord_name_error() -> None:
     [
         ('C', [('C', 0), ('E', 0), ('G', 0)]),
         ('C7', [('C', 0), ('E', 0), ('G', 0), ('Bb', 0)]),
-        ('Bbmaj7/D', [('Bb', 0), ('D', 1), ('F', 1), ('A', 1)]),
+        ('Bbmaj7/D', [('D', 0), ('F', 0), ('A', 0), ('Bb', 0)]),
     ]
 )
 def test_chord_name_to_chord(name: str, expected: list) -> None:
     chord_name = notes.ChordName(name)
     expected = notes.Chord([notes.Note(*n) for n in expected])
-    actual = chord_name.get_chord()
+    actual = chord_name.get_close_chord()
     assert actual == expected
 
 
 def test_chord_name_to_chord_different_lower() -> None:
-    actual = notes.ChordName('C').get_chord(lower=notes.Note('E', 2))
+    actual = notes.ChordName('C').get_close_chord(lower=notes.Note('E', 2))
     expected = notes.Chord([
         notes.Note('C', 3),
         notes.Note('E', 3),
         notes.Note('G', 3),
     ])
     assert actual == expected
+
+
+@pytest.mark.parametrize(
+    'note,other,allow_equal,octave',
+    [
+        (('C', 3), 'E', True, 3),
+        (('C', 3), 'C', True, 3),
+        (('C', 3), 'C', False, 4),
+        (('G', 3), 'D', True, 4),
+    ]
+)
+def test_nearest_above(note: tuple[str, int], other: str, allow_equal: bool, octave: int) -> None:
+    expected = notes.Note(other, octave)
+    actual = notes.Note(*note).nearest_above(other, allow_equal=allow_equal)
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    'note,other,allow_equal,octave',
+    [
+        (('C', 3), 'E', True, 2),
+        (('C', 3), 'C', True, 3),
+        (('C', 3), 'C', False, 2),
+        (('G', 3), 'D', True, 3),
+    ]
+)
+def test_nearest_above(note: tuple[str, int], other: str, allow_equal: bool, octave: int) -> None:
+    expected = notes.Note(other, octave)
+    actual = notes.Note(*note).nearest_below(other, allow_equal=allow_equal)
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    'l,n,expected',
+    [
+        ([1, 2, 3, 4], 0, [1, 2, 3, 4]),
+        ([1, 2, 3, 4], 1, [2, 3, 4, 1]),
+        ([1, 2, 3, 4], 3, [4, 1, 2, 3]),
+        ([1, 2, 3, 4], 5, None),
+    ]
+)
+def test_rotate_list(l: list[int], n: int, expected: list[int]) -> None:
+
+    if expected is None:
+        with pytest.raises(ValueError):
+            notes._rotate_list(l, n)
+    else:
+        actual = notes._rotate_list(l, n)
+        assert actual == expected
```

Comment:
Not 100% sure you need to test private functions which aren't part of the public facing API.

Reply:
It was the easiest way I could convince myself it was doing what I wanted it to 😅